zfs-replicate.sh

```diff
@@ -1,524 +1,538 @@
+  clear_lock "${LOGBASE}"/.push.lock
+  clear_lock "${LOGBASE}"/.pull.lock
+  clear_lock "${LOGBASE}"/.snapshot.lock
+
+  ## print log message and exit
+  printf "%s\n" "$logMsg"
+  exit 0
 }
 
 ## lockfile creation and maintenance
 check_lock() {
-    ## check our lockfile status
-    if [ -f "${1}" ]; then
-        ## get lockfile contents
-        local lpid=$(cat "${1}")
-        ## see if this pid is still running
-        local ps=$(ps auxww|grep $lpid|grep -v grep)
-        if [ "${ps}x" != 'x' ]; then
-            ## looks like it's still running
-            printf "ERROR: This script is already running as: %s\n" "${ps}"
-        else
-            ## well the lockfile is there...stale?
-            printf "ERROR: Lockfile exists: %s\n" "${1}"
-            printf "However, the contents do not match any "
-            printf "currently running process...stale lockfile?\n"
-        fi
-        ## tell em what to do...
-        printf "To run script please delete: %s\n" "${1}"
-        ## compress log and exit...
-        exit_error
+  local lockFile=$1
+  ## check our lockfile status
+  if [ -f "$lockFile" ]; then
+    ## see if this pid is still running
+    local ps
+    if ps=$(pgrep -lx -F "$lockFile"); then
+      ## looks like it's still running
+      printf "ERROR: This script is already running as: %s\n" "$ps"
     else
-        ## well no lockfile..let's make a new one
-        printf "Creating lockfile: %s\n" "${1}"
-        echo $$ > "${1}"
+      ## well the lockfile is there...stale?
+      printf "ERROR: Lockfile exists: %s\n" "$lockFile"
+      printf "However, the contents do not match any "
+      printf "currently running process...stale lockfile?\n"
     fi
+    ## cleanup and exit
+    exit_clean 99 "To run script please delete: $lockFile"
+  else
+    ## well no lockfile..let's make a new one
+    printf "Creating lockfile: %s\n" "$lockFile"
+    echo $$ > "$lockFile"
+  fi
 }
 
 ## delete lockfiles
 clear_lock() {
-    ## delete lockfiles...and that's all we do here
-    if [ -f "${1}" ]; then
-        printf "Deleting lockfile: %s\n" "${1}"
-        rm "${1}"
-    fi
+  local lockFile=$1
+  ## delete lockfiles...and that's all we do here
+  if [ -f "$lockFile" ]; then
+    printf "Deleting lockfile: %s\n" "$lockFile"
+    rm -f "$lockFile"
+  fi
 }
 
 ## check remote system health
 check_remote() {
-    ## do we have a remote check defined
-    if [ "${REMOTE_CHECK}x" != 'x' ]; then
-        ## run the check
-        $REMOTE_CHECK > /dev/null 2>&1
-        ## exit if above returned non-zero
-        if [ $? != 0 ]; then
-            printf "ERROR: Remote health check '%s' failed!\n" "${REMOTE_CHECK}"
-            exit_error
-        fi
-    fi
+  ## do we have a remote check defined
+  if [ "${REMOTE_CHECK}" == "" ]; then
+    return
+  fi
+  ## run the check
+  if ! $REMOTE_CHECK > /dev/null 2>&1; then
+    exit_clean $? "Remote check '${REMOTE_CHECK}' failed!"
+  fi
 }
 
 ## push replication function
 do_push() {
```

Comment:
In the `do_push` function `remote_set` is never defined, but it's referenced. This will just be an empty string.

Reply:
I never touched this code as far as I know. But what I was trying to do in relation to the remote check was to not error out if the remote check fails AND the MODE was set to LOCAL as their wouldn't be a remote check in that case.